src/__tests__/index.test.ts

```diff
@@ -0,0 +1,481 @@
+        beforeEach(function () {
+          aliasedWrapper = new OracleWrapper(
+            dummy.dbCredentials,
+            { alias: testAlias }
+          );
+        });
+        it('Should use the alias as the property', function () {
+          // eslint-disable-next-line @typescript-eslint/dot-notation
+          strictEqual(aliasedWrapper['alias'], testAlias);
+        });
+      });
+      context('Without a value in options', function () {
+        beforeEach(function () {
+          aliasedWrapper = new OracleWrapper(dummy.dbCredentials);
+        });
+        it('Should use the sid as the alias property', function () {
+          // eslint-disable-next-line @typescript-eslint/dot-notation
+          strictEqual(aliasedWrapper['alias'], dummy.dbCredentials.sid);
+        });
+      });
+    });
+  });
+  describe('getConnection', function () {
+    context('When a connection pool hasn\'t been instantiated', function () {
+      context('When createPool fails', function () {
+        let testError: Error;
+        beforeEach(function () {
+          testError = new Error('createPool failed');
+          createPoolStub.rejects(testError);
+        });
+        it('Should attempt to create a new connection pool', async function () {
+          try {
+            await wrapper.getConnection();
+            fail('Did not throw an error');
+          } catch (error) {
+            strictEqual(createPoolStub.callCount, 1);
+          }
+        });
+        it('Should use the alias from options', async function () {
+          try {
+            await wrapper.getConnection();
+            fail('Did not throw an error');
+          } catch (error) {
+            const { poolAlias } = createPoolStub.args[0][0] as PoolAttributes;
+            strictEqual(poolAlias, options.alias);
+          }
+        });
+        it('Should use the credentials', async function () {
+          try {
+            await wrapper.getConnection();
+            fail('Did not throw an error');
+          } catch (error) {
+            const {
+              user,
+              password,
+              connectionString,
+            } = createPoolStub.args[0][0] as PoolAttributes;
+            strictEqual(user, dummy.user);
+            strictEqual(password, dummy.password);
+            strictEqual(
+              connectionString,
+              `${dummy.host}:${dummy.port}/${dummy.sid}`
+            );
+          }
+        });
+        it('Should log an error notice', async function () {
+          try {
+            await wrapper.getConnection();
+            fail('Did not throw an error');
+          } catch (error) {
+            strictEqual(testLogger.error.callCount, 2);
+            strictEqual(
+              testLogger.error.args[0][0],
+              'Failed to create connectionPool'
+            );
+            strictEqual(testLogger.error.args[1][0], testError);
+          }
+        });
+        it('Should throw the error', async function () {
```

Comment:
Should we also test in here (or in another test case) that the error thrown is logged to `logger.error`?

Reply:
Ignore that. I somehow missed the test case above